chall03/ssacrist.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python3
+
+import sys
+
+def byebye():
+    print("usage: %s <1-9 squared_rows...>" % sys.argv[0])
+    sys.exit()
+
+def check():
+    if len(sys.argv) < 2 or len(sys.argv) > 21:
+        byebye()
+    sides = (len(sys.argv)) - 1
+    long = (len(sys.argv[1]))
+    if sides != long:
+        byebye()
+    for i in range(2, sides+1):
+        if len(sys.argv[i]) != long or (sys.argv[i]).isdigit() == False:
+            byebye()
+
+def main():
+    check()
+    matrix = []
+    for i in range(1, len(sys.argv)):
+        matrix.append(sys.argv[i])
+    i = 0
+    j = 0
+    c = (len(sys.argv)) - 1
+    h = 1
+    print(matrix[i][j], end='')
+    while c > 0:
+        for contador in range (h, c):
+            j += 1
+            print(", %s" %(matrix[i][j]), end='')
+        c -= 1
+        for contador in range (0, c):
+            i += 1
+            print(", %s" %(matrix[i][j]), end='')
+        for contador in range (c, 0, -1): 
+            j -= 1
+            print(", %s" %(matrix[i][j]), end='')
+        c -= 1
+        for contador in range (c, 0, -1):
+            i -= 1
+            print(", %s" %(matrix[i][j]), end='')
+        h = 0
+    print('')
```

Comment:
This looks too much like C. You should take a look at your colleague's solutions (the ones who didn't cheat) for a more "pythonic" approach 👍